client/catalog/src/app/datatypes/DatatypeDetail.svelte

```diff
@@ -0,0 +1,70 @@
+<script>
+export let params;
+import client from '../../data/client.js';
+import {Runtime, Storage} from '../../data/enums.js';
+import { breadcrumbStore } from '../stores.js';
+import {cap} from '../util.js';
+import { onMount } from 'svelte';
+import HLRowButton from '../../ui/HLRowButton.svelte';
+import HLTable from '../../ui/HLTable.svelte';
+import HLButton from '../../ui/HLButton.svelte';
+import HLRow from '../../ui/HLRow.svelte';
+import HLTextBig from '../../ui/HLTextBig.svelte';
+import HLSelect from '../../ui/HLSelect.svelte';
+import CodeMirror from 'codemirror';
+import 'codemirror/mode/python/python.js';
+
+let id = params.id;
+let load = client.datatype.findOne({where: {id: id}, include: {validator: true}});
+
+var cm;
+onMount(async () => {
+	load.then(obj => {
+		breadcrumbStore.set(
+			[{
+				href: "/datatypes",
+				text: "Datatypes",
+			}, {
+				href: "/datatypes/" + id,
+				text: cap(obj.name),
+			}]
+		);
+		if(obj.validator.runtime == 2) {
+			cm = CodeMirror.fromTextArea(document.querySelector('#code'), {
+				mode: {name:"python"},
+				lineNumbers: true,
+				indentUnit: 4,
+				theme: "duotone-dark",
+				readOnly: true,
+			});
+			cm.setValue(obj.validator.code);
+		}
+	});
+});
+</script>
+
+<style>
+	.box {
+		margin: 1em 1.5em; 
+	}
+</style>
+
+<div class="box">
+	{#await load}
+		&nbsp;
+	{:then datatype}
+	<h1>{cap(datatype.name)}</h1>
+	<HLTable>
+		<h2>Runtime: {Runtime[datatype.validator.runtime]}</h2>
+	</HLTable>
+	<HLTable>
+		<textarea id="code" name="code" style="display: none;"></textarea>
```

Comment:
I wonder if there's a way to make this look natural and take up much more of the screen.. or if there's a separate view/edit UI or something. Scrolling around the box now is a bit awkward.

<img width="1372" alt="Screen Shot 2020-06-14 at 7 19 13 AM" src="https://user-images.githubusercontent.com/55026/84595912-72ae7600-ae0f-11ea-9e88-4e1c2205a2e1.png">

Reply:
Just removing the hltable so it takes the whole width. Easy change!